docs/deep-dives/mobile/enrollment-state-machine.mdx

```diff
@@ -0,0 +1,176 @@
+---
+id: mobile-enrollment-state-machine
+title: Enrollment State Machine
+slug: /mobile-enrollment-state-machine
+---
+
+## Possible States
+
+Within the Nimbus SDK there are multiple states, most of which have a "reason" associated with them.
+The reason field functions as a child state, and is required for parent states that have reasons.
+
+### Enrolled
+
+#### Qualified
+
+The client is not already enrolled in a recipe of the same type that would result in a feature conflict, and matched targeting and bucketing criteria for this recipe<sup><a href="#notes">[1]</a></sup>.
+
+#### Opt In
+
+The client explicitly opted into this recipe<sup><a href="#notes">[2]</a></sup>.
```

Comment:
As far as I'm aware the CLI actually modifies targeting/bucketing criteria, resulting in the `Qualified` state rather than opt-in. This opt-in is referring to a method present on the NimbusClient that's only used in the secret menu as far as I can tell.

Reply:
> the CLI … modifies targeting/bucketing criteria, resulting in the `Qualified` state 

This is correct.